File: go/test/protocol/test_case.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

type TestCase struct {
	TestName    string              `yaml:"test-name"`
	Description TestCaseDescription `yaml:"description"`
	Requires    []TestFeatureKind   `yaml:"requires"`
	Prologue    TestCasePrologue    `yaml:"prologue"`
	Actions     []TestCaseAction    `yaml:"actions"`
	Epilogue    TestCaseEpilogue    `yaml:"epilogue"`
}

File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCase.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCase
    {
        public string? TestName { get; set; }

        public TestCaseDescription? Description { get; set; }

        public List<TestFeatureKind> Requires { get; set; } = new();

        public TestCasePrologue? Prologue { get; set; }

        public List<TestCaseAction> Actions { get; set; } = new();

        public TestCaseEpilogue? Epilogue { get; set; }
    }
}
